Modules/_asynciomodule.c

```diff
@@ -1371,10 +1371,15 @@ _asyncio_Future__make_cancelled_error_impl(FutureObj *self)
 {
     PyObject *exc = create_cancelled_error(self->fut_cancel_msg);
     _PyErr_StackItem *exc_state = &self->fut_cancelled_exc_state;
-    /* Transfer ownership of exc_value from exc_state to exc since we are
-       done with it. */
-    PyException_SetContext(exc, exc_state->exc_value);
-    exc_state->exc_value = NULL;
```

Comment:
Note that PyErr_Occurred checks whether the type is NULL, not the value. So setting just the value to NULL here doesn't follow the interpreter's semantics for the triplet.